### PR TITLE
registry: mirroring of blobs

### PR DESCRIPTION
**Pull request by miminar**
1.5 release.

### CODE REVIEW

**Comment by miminar** [comment]
@legionus PTAL, thanks

File: install_config/registry/extended_registry_configuration.adoc [hunk 1]
```diff
@@ -362,23 +362,30 @@ middleware:
...
 link:https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md#image-manifest-version-2-schema-2[manifest
-schema v2] during a push to the registry. See xref:middleware-repository-acceptschema2[below] for more details.
-<3> Let the registry act as a proxy for remote blobs. See xref:middleware-repository-pullthrough[below] for more details.
-<4> Prevent blob uploads exceeding size limit defined in targeted project.
-<5> An expiration timeout for limits cached in the registry. The lower the
+schema v2] during a push to the registry. See
+xref:middleware-repository-acceptschema2[below] for more details.
+<3> Let the registry act as a proxy for remote blobs. See
+xref:middleware-repository-pullthrough[below] for more details.
+<4> Let the registry cache blobs served from remote registries for later fast
```
Comment (legionus) [inline install_config/registry/extended_registry_configuration.adoc]: @miminar Can you mention that mirroring starts only after the first access to blob ?

Comment (miminar) [inline install_config/registry/extended_registry_configuration.adoc]: Good point.

Comment (miminar) [inline install_config/registry/extended_registry_configuration.adoc]: @legionus Addressed.

**Review by legionus — APPROVED** [review]
LGTM

**Comment by miminar** [comment]
@ahardin-rh may I have your review here as well?

File: install_config/registry/extended_registry_configuration.adoc [hunk 2]
```diff
@@ -362,23 +362,31 @@ middleware:
...
+access. The mirroring starts when the blob is accessed for the first time. The
+option has no effect if the xref:middleware-repository-pullthrough[pullthrough]
+is disabled.
+<5> Prevent blob uploads exceeding size limit defined in targeted project.
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: s/Prevent/Prevents
s/exceeding size/exceeding the size
s/in targeted/in the targeted

File: install_config/registry/extended_registry_configuration.adoc [hunk 3]
```diff
@@ -362,23 +362,31 @@ middleware:
...
+        mirrorpullthrough: true <4>
+        enforcequota: false <5>
+        projectcachettl: 1m <6>
+        blobrepositorycachettl: 10m <7>
   storage:
     - name: openshift <1>
 ----
-<1> These entries are mandatory. Their presence ensures required components get loaded. These values shouldn't be changed.
+<1> These entries are mandatory. Their presence ensures required components get
+loaded. These values shouldn't be changed.
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: s/shouldn't/should not

File: install_config/registry/extended_registry_configuration.adoc [hunk 4]
```diff
@@ -505,6 +522,23 @@ this way, will not be stored in the registry.
...
+of this mirroring feature is an increased storage usage.
+
+[NOTE]
+====
+The mirroring starts when a client tries to fetch at least a single byte of the
+blob. To pre-fetch a particular image into integrated registry before it is
+actually needed, you can run the following command:
+
+----
+oc get imagestreamtag/${IS}:${TAG} -o jsonpath='{ .image.dockerImageLayers[*].name }' | \
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: $ oc

**Comment by miminar** [comment]
Addressed. Thanks

**Comment by ahardin-rh** [comment]
@miminar Looks like we have a build failure with Travis. Do you mind rebasing to see if that resolves it?

**Comment by miminar** [comment]
@ahardin-rh No problem. Rebased.

**Comment by ahardin-rh** [comment]
Thank you!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/registry/extended_registry_configuration.adoc#install-config-registry-extended-configuration[Setting up the Registry -> Extended Registry Configuration]
|Added details about mirroring of blobs.
%

**Pull request merged**